e2e/playwright/tests/@gitops/test.spec.ts

```diff
@@ -0,0 +1,230 @@
+  }
+
+  // get the key ID from the response
+  const data = await response.json();
+  const keyId = data.id;
+  // at the end of the test, or on failure, reset the GitHub repo
+  // TODO: figure out how to do this
+  // test.afterAll(async () => {
+  //   await resetGithub(githubToken, gitopsOwner, gitopsRepo, keyId, randomBranch);
+  // });
+
+  // enable gitops now that the key is added
+  await page.getByText('Test connection to repository').click();
+  await expect(page.getByText('GitOps is enabled')).toBeVisible();
+  await page.getByText('Go to dashboard').click();
+  await page.getByText('Version history').click();
+  await expect(page.getByText('Deploy')).not.toBeVisible();
+  await expect(page.getByText('Redeploy')).not.toBeVisible();
+  // wait 10 seconds for things to actually be committed
+  await page.waitForTimeout(10000);
+  await trivialConfig(page, true);
+
+  const commitResponse = await fetch(`https://api.github.com/repos/${gitopsOwner}/${gitopsRepo}/commits?sha=${randomBranch}`, {
+    headers: {
+      'Authorization': `token ${githubToken}`,
+      'Accept': 'application/vnd.github.v3+json'
+    }
+  })
+  if (commitResponse.status !== 200) {
+    throw new Error(`Failed to get commits from GitHub. Status: ${commitResponse.status} Contents: ${await commitResponse.text()}`);
+  }
+  const commitResponseJson = await commitResponse.json();
+  // there should be 3 commits - one for the initial version, and then two for the two config changes
+  checkCommitMessageExists(commitResponseJson, 'Updating GitOps to version 2')
+  checkCommitMessageExists(commitResponseJson, 'Updating GitOps to version 1')
+  checkCommitMessageExists(commitResponseJson, 'Updating GitOps to version 0')
+
+  const contentResponse = await fetch(`https://api.github.com/repos/${gitopsOwner}/${gitopsRepo}/contents/${randomPath}/${testAppSlug}.yaml?ref=${randomBranch}`, {
+    headers: {
+      'Authorization': `token ${githubToken}`,
+      'Accept': 'application/vnd.github.v3+json'
+    }
+  })
+  if (contentResponse.status !== 200) {
+    throw new Error(`Failed to get content from GitHub. Status: ${contentResponse.status} Contents: ${await contentResponse.text()}`);
+  }
+  const contentResponseJson = await contentResponse.json();
+  checkFilePathExists(contentResponseJson, `${randomPath}/${testAppSlug}.yaml`)
+
+  // turn off gitops so we can test the behavior of kots afterwards
+  await resetGithub(githubToken, gitopsOwner, gitopsRepo, keyId, randomBranch);
+
+  await page.locator('span').filter({ hasText: 'GitOps' }).click();
+  await page.getByText('Disable GitOps for this app').click();
+  await expect(page.getByText('Are you sure you want to disable GitOps for this application?')).toBeVisible();
+  await page.getByRole('button', { name: 'Disable GitOps' }).click();
+  await expect(page.getByText('Are you sure you want to disable GitOps for this application?')).not.toBeVisible();
+  await expect(page.getByText('Not Enabled')).toBeVisible();
```

Comment:
this seems like the type of text that might exist somewhere else on the page causing this to be flakey. how about using a data-testid?